Automattic/wp-calypso
# Plans: Move useStorageAddOns logic  selection in pricing meta hook

Related to https://github.com/Automattic/wp-calypso/pull/96520#discussion_r1848660432

## Proposed Changes

- Query AddOns.useStorageAddOns hook internally within `UsePricingMetaForGridPlans` hook
- Replace `storageAddOns` param in `UsePricingMetaForGridPlans` hook with `reflectStorageSelectionInPlanPrices`
- Fix any associated tests

## Why are these changes being made?


Storage add-ons have become a concrete feature supported by the plans grid package. We can simplify the interface of the `usePricingMetaForGridPlans` hook by handling storage add-on data retrieval within the hook itself, rather than expecting consumers to pass in the relevant data.

## Testing Instructions



* Visit `/start/plans`
* Select different storage add-on options for annual Business and Ecommerce plans
* Verify that the header pricing and billing timeframe update accordingly
* Repeat for the logged in plans page
* Go to /plans/:site?flags=plans/term-savings-price-display and /start/plans?flags=plans/term-savings-price-display
* Ensure that when storage add-ons are selected for business or ecommerce annual plans, the header price doesn't change
* Smoke test several other instances where the `usePricingMetaForGridPlans` is used

## Pre-merge Checklist
